docs/development-guide.md

```diff
@@ -0,0 +1,192 @@
+# Development Guide
+
+## New to Go
+
+Ceph-csi is written in Go and if you are new to the language,
+it is **highly** encouraged to:
+
+* Take the [A Tour of Go](http://tour.golang.org/welcome/1) course.
+* [Set up](https://golang.org/doc/code.html) Go development environment on your machine.
+* Read [Effective Go](https://golang.org/doc/effective_go.html) for best practices.
+
+## Development Workflow
+
+### Workspace and repository setup
+
+* [Download](https://golang.org/dl/) Go and
+   [install](https://golang.org/doc/install) it on your system.
+* Run
+
+    ```console
+    go get -d github.com/ceph/ceph-csi-operator
+    ```
+
+   This will just download the source and not build it. The downloaded source
+   will be at `$GOPATH/src/github.com/ceph/ceph-csi-operator`
+* Fork the [ceph-csi-operator repo](https://github.com/ceph/ceph-csi-operator) on Github.
+* Add your fork as a git remote:
+
+    ```console
+    git remote add fork https://github.com/<your-github-username>/ceph-csi-operator
+    ```
+
+> Editors: Our favorite editor is vim with the [vim-go](https://github.com/fatih/vim-go)
+> plugin, but there are many others like [vscode](https://github.com/Microsoft/vscode-go)
+
+### Building ceph-csi-operator
+
+To build ceph-csi-operator locally run:
+
+```console
+make
+```
+
+To build ceph-csi-operator in a container:
+
+```console
+make docker-build
+```
+
+The built binary will be present under `bin/` directory.
+
+### Running ceph-csi-operator tests
+
+Once the changes to the sources compile, it is good practise to run the tests
+that validate the style and other basics of the source code. Execute the unit
+tests (in the `*_test.go` files) and check the formatting of YAML files,
+MarkDown documents and shell scripts:
+
+```console
+make test
+```
+
+In addition to running tests locally, each Pull Request that is created will
+trigger Continuous Integration tests, but
+also additional functionality tests that are defined under the `e2e/`
+directory.
+
+### Code contribution workflow
+
+ceph-csi-operator repository currently follows GitHub's
+[Fork & Pull] (<https://help.github.com/articles/about-pull-requests/>) workflow
+for code contributions.
+
+Please read the [coding guidelines](coding.md) document before submitting a PR.
+
+#### Certificate of Origin
+
+By contributing to this project you agree to the Developer Certificate of
+Origin (DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution. See the [DCO](DCO) file for details.
+
+Contributors sign-off that they adhere to these requirements by adding a
+Signed-off-by line to commit messages. For example:
+
+```
+subsystem: This is my commit message
```

Comment:
@Madhu-1 we need to discuss if we should follow the `subsystem` methodology in this repo. I am not an advocate, but I am open to discussion if we can show the benefits of using such methodology

Reply:
Sure, lets discuss that in meeting